src/mesh/mesh_base.C

```diff
@@ -515,4 +521,105 @@ void MeshBase::cache_elem_dims()
   this->comm().set_union(_elem_dims);
 }
 
+void MeshBase::detect_interior_parents()
+{
+  // This requires an inspection on every processor
+  parallel_object_only();
+
+  if( this->elem_dimensions().size() == 1 )
```

Comment:
Let's add a comment here indicating why we do this, i.e. we only need to do this check if the mesh has elements of multiple different dimensions.